src/Compilers/Core/CodeAnalysisTest/EditorConfigTests.cs

```diff
@@ -394,6 +394,133 @@ public void LiteralQuestions()
             Assert.DoesNotMatch(regex, "/?axcde?f");
         }
 
+        [Fact]
+        public void LiteralBraces()
+        {
+            string regex = EditorConfig.TryCompileSectionNameToRegEx("abc\\{\\}def");
+            Assert.Equal("^.*/abc\\{\\}def$", regex);
+
+            Assert.Matches(regex, "/abc{}def");
+            Assert.Matches(regex, "/subdir/abc{}def");
+            Assert.DoesNotMatch(regex, "/abcdef");
+            Assert.DoesNotMatch(regex, "/abc}{def");
+        }
+
+        [Fact]
+        public void LiteralComma()
+        {
+            string regex = EditorConfig.TryCompileSectionNameToRegEx("abc\\,def");
+            Assert.Equal("^.*/abc,def$", regex);
+
+            Assert.Matches(regex, "/abc,def");
+            Assert.Matches(regex, "/subdir/abc,def");
+            Assert.DoesNotMatch(regex, "/abcdef");
+            Assert.DoesNotMatch(regex, "/abc\\,def");
+            Assert.DoesNotMatch(regex, "/abc`def");
+        }
+
+        [Fact]
+        public void SimpleChoice()
+        {
+            string regex = EditorConfig.TryCompileSectionNameToRegEx("*.{cs,vb,fs}");
+            Assert.Equal("^.*/[^/]*\\.(?:cs|vb|fs)$", regex);
+
+            Assert.Matches(regex, "/abc.cs");
+            Assert.Matches(regex, "/abc.vb");
+            Assert.Matches(regex, "/abc.fs");
+            Assert.Matches(regex, "/subdir/abc.cs");
+            Assert.Matches(regex, "/subdir/abc.vb");
+            Assert.Matches(regex, "/subdir/abc.fs");
+
+            Assert.DoesNotMatch(regex, "/abcxcs");
```

Comment:
Consider adding a test with a `.(cs|vb|fs)` in the middle of the string to ensure that we're matching to the end of the string.